fvm/Grid/AnalyticBRadialGridGenerator.cpp

```diff
@@ -358,36 +358,342 @@ void AnalyticBRadialGridGenerator::InterpolateInputProfileToGrid(
 }
 
 /**
- * Calculate minor radius coordinate 'r' corresponding to the given
- * Cartesian coordinates (x,y,z).
+ * Interpolates the input chaping profile at a single radius r, and extrapolates using 
+ * the maximum and minimum r for which the profile is defined. This is useful for the 
+ * GetRThetaPhiFromCartesian() and GetGradRCartesian() functions used for e.g. the SPI
+ * module in DREAM, as these functions must be able to handle coordinates outside the plasma
  *
- * (The Cartesian coordinate system is oriented such that x and y span
- * the poloidal plane. The origin of x and y is the magnetic axis.)
  */
-void AnalyticBRadialGridGenerator::GetRThetaFromCartesian(real_t* /*r*/, real_t* /*theta*/,
-    real_t /*x*/, real_t /*y*/, real_t /*z*/, real_t /*lengthScale*/, real_t /*startingGuessR*/
+real_t AnalyticBRadialGridGenerator::InterpolateInputProfileSingleExtrap(real_t r,
```

Comment:
Wouldn't it be nice to implement helper methods for interpolating specifically in the various shaping parameters which utilize this method? E.g. a method ``InterpolateInputElongation(real_t r)``, which just returns ``kappa`` in the requested point ``r`` (and same for the other shaping parameters).